### PR TITLE
[TRAFODION-2775] Insert does not raise duplicate row error for hbase …

### PR DESCRIPTION
**Pull request by sureshsubbiah**
…format table with defaulted first column

There was a problem with column being checked during CheckAndPut call for such tables. We were always checking the first column in the row, which may not have existed in the previous version of this rowid. Please see JIRA for a complete explanation.

### CODE REVIEW

File: core/sql/exp/ExpHbaseInterface.cpp [hunk 1]
```diff
@@ -842,7 +842,9 @@ Lng32 ExpHbaseInterface_JNI::insertRow(
     transID = getTransactionIDFromContext();
   retCode_ = client_->insertRow((NAHeap *)heap_, tblName.val, hbs_,
                                 useTRex_, transID, rowID, row, timestamp, 
-                                checkAndPut, asyncOperation, useRegionXn, &htc);
+                                checkAndPut, asyncOperation, useRegionXn, 
+				0, // checkAndPut is false, so colIndexToCheck is not used
```
Comment (DaveBirdsall) [inline core/sql/exp/ExpHbaseInterface.cpp]: How will the insertRow method distinguish between the "not used" case and a column index of zero? (Elsewhere I saw a comment that the column index is 0-based.)

Comment (sureshsubbiah) [reply core/sql/exp/ExpHbaseInterface.cpp]: The java code in HTableClient::putRow has this line of code which will ensure that the number passed in for colIndexToCheck is meaningless when checkAndPut is FALSE. -1 would have been easier to read. I can make that change later, if you agree.

if (checkAndPut && colIndex == colIndexToCheck) {

Comment (DaveBirdsall) [reply core/sql/exp/ExpHbaseInterface.cpp]: Thanks; code change looks good to me.

**Comment by selvaganesang** [comment]
In the HTableClient.putRow method there is a parameter columnToCheck which expects the column name instead of colIndex. To avoid confusion, please consider removing this parameter.

**Pull request merged**